### ISSUE TITLE
0.21.3: build fails

### ISSUE THREAD

**Issue by kloczek** [rockdaboot/libpsl#222]
<details>
<summary>Looks like build fails</summary>

```console
[1/10] Compiling C object src/libpsl.so.5.3.5.p/psl.c.o
FAILED: src/libpsl.so.5.3.5.p/psl.c.o
/usr/bin/gcc -Isrc/libpsl.so.5.3.5.p -Isrc -I../src -I. -I.. -Iinclude -I../include -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -fPIC -DHAVE_CONFIG_H -DBUILDING_PSL '-DPSL_DISTFILE="/usr/share/publicsuffix/public_suffix_list.dafsa"' -MD -MQ src/libpsl.so.5.3.5.p/psl.c.o -MF src/libpsl.so.5.3.5.p/psl.c.o.d -o src/libpsl.so.5.3.5.p/psl.c.o -c ../src/psl.c
../src/psl.c: In function ‘psl_str_to_utf8lower’:
../src/psl.c:108:26: error: ‘ICONV_CONST’ undeclared (first use in this function); did you mean ‘WINICONV_CONST’?
  108 | #  define WINICONV_CONST ICONV_CONST
      |                          ^~~~~~~~~~~
../src/psl.c:1838:53: note: in expansion of macro ‘WINICONV_CONST’
 1838 |                                 else if (iconv(cd, (WINICONV_CONST char **)&tmp, &tmp_len, &dst_tmp, &dst_len_tmp) != (size_t)-1
      |                                                     ^~~~~~~~~~~~~~
../src/psl.c:108:26: note: each undeclared identifier is reported only once for each function it appears in
  108 | #  define WINICONV_CONST ICONV_CONST
      |                          ^~~~~~~~~~~
../src/psl.c:1838:53: note: in expansion of macro ‘WINICONV_CONST’
 1838 |                                 else if (iconv(cd, (WINICONV_CONST char **)&tmp, &tmp_len, &dst_tmp, &dst_len_tmp) != (size_t)-1
      |                                                     ^~~~~~~~~~~~~~
../src/psl.c:1838:68: error: expected ‘)’ before ‘char’
 1838 |                                 else if (iconv(cd, (WINICONV_CONST char **)&tmp, &tmp_len, &dst_tmp, &dst_len_tmp) != (size_t)-1
      |                                                    ~               ^~~~
ninja: build stopped: subcommand failed.
```
</details>

<details>
<summary>meson output</summary>

```console
+ /usr/bin/meson setup --buildtype=plain --prefix=/usr --libdir=/usr/lib64 --libexecdir=/usr/libexec --bindir=/usr/bin --sbindir=/usr/sbin --includedir=/usr/include --datadir=/usr/share --mandir=/usr/share/man --infodir=/usr/share/info --localedir=/usr/share/locale --sysconfdir=/etc --localstatedir=/var --sharedstatedir=/var/lib --wrap-mode=nodownload --auto-features=enabled . x86_64-redhat-linux-gnu -D runtime=libidn2 -D builtin=false -D psl_distfile=/usr/share/publicsuffix/public_suffix_list.dafsa -D psl_file=/usr/share/publicsuffix/effective_tld_names.dat -D psl_testfile=/usr/share/publicsuffix/test_psl.txt
The Meson build system
Version: 1.3.1
Source dir: /home/tkloczko/rpmbuild/BUILD/libpsl-0.21.3
Build dir: /home/tkloczko/rpmbuild/BUILD/libpsl-0.21.3/x86_64-redhat-linux-gnu
Build type: native build
Project name: libpsl
Project version: 0.21.3
C compiler for the host machine: /usr/bin/gcc (gcc 13.2.1 "gcc (GCC) 13.2.1 20231205 (Red Hat 13.2.1-6)")
C linker for the host machine: /usr/bin/gcc ld.bfd 2.41-20
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.0
Run-time dependency libidn2 found: YES 2.3.4
Library unistring found: YES
Run-time dependency iconv found: YES
Check usable header "unistd.h" : YES
Check usable header "stdint.h" : YES
Check usable header "alloca.h" : YES
Check usable header "dirent.h" : YES
Checking for function "alloca" : YES
Checking for function "strndup" : YES
Checking for function "clock_gettime" : YES
Checking for function "fmemopen" : YES
Checking for function "nl_langinfo" : YES
Compiler for C supports function attribute visibility: YES
Configuring config.h using configuration
Program python3 found: YES (/usr/bin/python3)
Configuring libpsl.h using configuration
Build targets in project: 11

libpsl 0.21.3

  User defined options
    auto_features : enabled
    bindir        : /usr/bin
    buildtype     : plain
    datadir       : /usr/share
    includedir    : /usr/include
    infodir       : /usr/share/info
    libdir        : /usr/lib64
    libexecdir    : /usr/libexec
    localedir     : /usr/share/locale
    localstatedir : /var
    mandir        : /usr/share/man
    prefix        : /usr
    sbindir       : /usr/sbin
    sharedstatedir: /var/lib
    sysconfdir    : /etc
    wrap_mode     : nodownload
    builtin       : false
    psl_distfile  : /usr/share/publicsuffix/public_suffix_list.dafsa
    psl_file      : /usr/share/publicsuffix/effective_tld_names.dat
    psl_testfile  : /usr/share/publicsuffix/test_psl.txt
    runtime       : libidn2
```
</details>

**Comment by kloczek** [comment]
Just in case . I'm using gettext 0.22.4.

**Comment by rockdaboot** [comment]
Hm, it should be defined in `config.h`, but I only test with autotools.

**Comment by rockdaboot** [comment]
I created #223, so that we can catch such issues earlier.

**Comment by kloczek** [comment]
In the same build env build of libpsl 0.21.2 is OK.

**Comment by kloczek** [comment]
> I created #223, so that we can catch such issues earlier.

One sec will try to test that 😋

**Comment by rockdaboot** [comment]
Yes, it is because I tried to fix iconv() on Solaris (and some other obscure systems).
`ICONV_CONST` should always be set in `config.h`, but surely meson doesn't do it.

**Comment by kloczek** [comment]
Meson can collect detection results in config.h file as well.

**Comment by kloczek** [comment]
> In the same build env build of libpsl 0.21.2 is OK.

In bot cases (0.21.2 and 0.21.3) I've been using meson.

**Comment by rockdaboot** [comment]
It can be fixed in meson or in the code (just an #ifdef).
But meson is a second-class build system for me (doesn't mean it's bad or so).
I am personally very familiar with autotools and stick with it.

**Comment by rockdaboot** [comment]
Can you test #224 ?

**Comment by kloczek** [comment]
I understand .. 👍 
This issue is not-so-critical and I have few other more important things to do. Just please let me know if you will have any fix.
I would be glad to try to test before actual merge 😋

**Comment by rockdaboot** [comment]
I just tested with meson, looks good now.

**Comment by rockdaboot** [comment]
Thx, released v0.21.4 just now

**Comment by kloczek** [comment]
Thank you 👍 😄
